deeptime/markov/tools/estimation/sparse/count_matrix.py

```diff
@@ -60,7 +67,26 @@ def count_matrix_coo2_mult(dtrajs, lag, sliding=True, sparse=True, nstates=None)
     # feed into one COO matrix
     row = np.concatenate(rows)
     col = np.concatenate(cols)
-    data = np.ones(row.size)
+    ## choose option for including reweighting factors g and M
+    if reweighting_factors is None:
+        data = np.ones(row.size)
+    elif type(reweighting_factors) is tuple:
+        g_factors, M_factors = reweighting_factors
+        factors=[]
+        for g,M in zip(g_factors,M_factors):
+            if g.size > lag:
+                if sliding:
+                    m = M.cumsum()
+                    m[lag:] = m[lag:] - m[:len(m)-lag]
+                    m = m[(lag):]
+                    m = np.exp(-m)
+                    factors.append(g[0:-lag]*m)
+                else:
+                    raise NotImplementedError('Only the sliding scheme is implemented.')
```

Comment:
I would expect that this is checked sooner, eg, right below the `elif` in line 73.